cli/azd/pkg/input/previewer.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package input
+
+import (
+	"bufio"
+	"log"
+	"strings"
+	"sync"
+
+	tm "github.com/buger/goterm"
+)
+
+/* progressLog implements the io.Writer interface and writes bytes applying a framed style like
+*
+************************************************
+*  <header>
+*
+*  <displayTitle>
+*
+*  <prefix><logA
+*  <prefix><logB>
+*  <prefix><logC>
+*  <prefix><logD>
+*  <prefix><logN...>
+*
+*  ───────────────────────────────────────────
+*************************************************
+*
+* Notes:
+* - prefix can be used for setting indentation for the output.
+* - The line size for the title and the footer are calculated during Start(), filling the available space from the screen.
+* - <displayTitle> is generated as `<prefix>──── <title> ──────────────` where:
+*    - lines on the left and right of the title are calculated based on the available space on screen minus the prefix and
+*      title length. The left line represent 1/10 of the entire line, leaving the rest for the right line.
+*    - The less space on the screen, the shorter the lines will be. Lines won't be displayed if there is not enough room.
+* - The title and the logs are truncated with the symbol `...` at the end when it is bigger than the screen width.
+ */
+type progressLog struct {
```

Comment:
I'd recommend renaming either the component or the file so they both match.  Current filename is `previewer.go` and the struct is `progressLog`.